src/backend/utils/adt/json.c

```diff
@@ -338,12 +354,67 @@ datum_to_json(Datum val, bool is_null, StringInfo result,
 			break;
 		default:
 			outputstr = OidOutputFunctionCall(outfuncoid, val);
+			// The outfuncoid are specific to babelfish and therefore no macro is defined.
+			if ((sql_dialect==SQL_DIALECT_TSQL) && 
```

Comment:
1. We should not depend on sql_dialect.
2. Our policy is to minimize PG Engine's change. Let's introduce a hook here.